### PR TITLE
refactor: replace console.log with info, warn and error, introduce shared mock setup

### LINKED ISSUE #277: Console mocking
In most tests, the console can be mocked. The `spyOn` will be used and implemented in a shared setup for all tests.

## Implementation details
The convention is:
- We use `console.error`, `console.warn` and `console.info` for expected information into the console.
- `console.log` is used for debugging.

### PR DESCRIPTION
**Pull request by Tlacenka**
Closes #277 

In this PR, I go through all packages and replace all occurrences of `console.log` with `console.info`, `console.warn` or `console.error`.

Additionally, I set up all `vitest` configuration files for both unit and integration tests to use console mocks.
- Exception: `multi-progress-bars` changes the behaviour of `console` [see here](https://github.com/kamiyo/multi-progress-bars/blob/5fa31e07930d3140b5f5c1aca78caa3ad34bd435/src/virtual-console.ts#L76). Currently, none of the utils integration tests use a console so until I get to that package refactor, I simply did not set up console mocks for integration tests there. But we may have to **create a special type of a test file** for such cases. What do you think?

### CODE REVIEW

File: packages/core/src/lib/implementation/persist.integration.test.ts [hunk 1]
```diff
@@ -56,29 +53,19 @@ const resetFiles = async () => {
...
 describe('persistReport', () => {
   beforeEach(async () => {
     resetFiles();
-    setupConsole();
   });
 
-  afterEach(() => {
-    teardownConsole();
-  });
+  afterEach(() => {});
```
Comment (matejchalk) [inline packages/core/src/lib/implementation/persist.integration.test.ts:63]: Can now be removed :slightly_smiling_face: 

```suggestion
```

**Review by matejchalk — COMMENTED** [review]
Excellent, great to have this in place :+1: Could also make sure the `no-console` lint rule is no longer disabled in any project? :pray:

File: packages/core/src/lib/implementation/persist.integration.test.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/core/src/lib/implementation/persist.integration.test.ts]: Oops 😄

**Comment by BioPhoton** [comment]
🔥

**Review by matejchalk — APPROVED** [review]
:tada:

**Pull request merged**